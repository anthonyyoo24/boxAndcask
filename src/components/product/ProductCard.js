import './ProductCard.scss';
import React from 'react';
import { Link } from 'react-router-dom';

const ProductCard = ({ product }) => {
  const image = require(`../../img/${product.image}`);

  return (
    <div key={product.id} className="product-card">
      <Link to={`/products/${product.id}`}>
        <img src={image} alt={product.name} className="product-card__img" />
      </Link>
      <div className="product-card__content">
        <Link to={`/products/${product.id}`} className="product-card__name">
          {product.name}
        </Link>
        <p className="product-card__price">${product.price}</p>
      </div>
    </div>
  );
};

export default ProductCard;
