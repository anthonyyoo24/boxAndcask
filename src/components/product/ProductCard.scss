@import '../../sass/base.scss';
@import '../../sass/variables.scss';

.product-card {
  border-radius: 1.6rem;
  overflow: hidden;
  box-shadow: $shadow-double-1of5;
  background-color: #ffffff;
  transition: all 0.5s;

  &:hover {
    box-shadow: $shadow-double-3of5;
  }

  &__img {
    max-width: 100%;
    border-bottom: 1px solid $blue-grey-light-1of5;
  }

  &__content {
    padding: 0.8rem;
  }

  &__name {
    color: $blue-grey-dark-5of5;
  }

  &__price {
    margin-top: 0.4rem;
    font-weight: 700;
  }
}
