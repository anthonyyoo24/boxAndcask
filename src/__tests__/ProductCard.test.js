import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import checkPropTypes from 'check-prop-types';
import ProductCard from '../components/product/ProductCard';

const fakeProduct = {
  name: 'Josephine Sofa',
  image: 'josephine-sofa.jpg',
  price: 1200.99,
  id: 2,
};

describe('ProductCard', () => {
  describe('Checking PropTypes', () => {
    it('should not throw a warning', () => {
      const mockProps = {
        key: 1,
        product: fakeProduct,
      };

      const propsError = checkPropTypes(ProductCard.propTypes, mockProps, 'prop', ProductCard.name);

      expect(propsError).toBeUndefined();
    });
  });

  describe('With props', () => {
    let productCardWrapper;
    beforeAll(() => {
      productCardWrapper = shallow(<ProductCard product={fakeProduct} />);
    });

    it('renders without error', () => {
      expect(productCardWrapper.find('.product-card')).toHaveLength(1);
    });

    it('renders the price', () => {
      const price = productCardWrapper.find('.product-card__price');

      expect(price).toHaveLength(1);
      expect(price.text()).toBe(`$${fakeProduct.price}`);
    });

    it('renders the image', () => {
      const img = productCardWrapper.find('img');

      expect(img).toHaveLength(1);
      expect(img.props().src).toBe(`img/${fakeProduct.image}`);
      expect(img.props().alt).toBe(fakeProduct.name);
      expect(img.props().className).toBe('product-card__img');
    });

    it('renders the links', () => {
      const links = productCardWrapper.find('Link');

      expect(links).toHaveLength(2);
      expect(links.find('img').exists()).toBe(true);
      expect(links.at(0).props().to).toBe(`/products/${fakeProduct.id}`);
      expect(links.at(1).props().to).toBe(`/products/${fakeProduct.id}`);
      expect(links.at(1).props().className).toBe('product-card__name');
      expect(links.at(1).text()).toBe(fakeProduct.name);
    });

    it('matches the snapshot', () => {
      expect(toJSON(productCardWrapper)).toMatchSnapshot();
    });

    it('updates via props', () => {
      const productCardWrapper = shallow(<ProductCard product={{ ...fakeProduct, price: 1000 }} />);
      expect(toJSON(productCardWrapper)).toMatchSnapshot();
      productCardWrapper.setProps({ product: { ...fakeProduct, price: 1100.99 } });
      expect(toJSON(productCardWrapper)).toMatchSnapshot();
    });
  });

  describe('Without props', () => {
    let productCardWrapper;
    beforeAll(() => {
      productCardWrapper = shallow(<ProductCard />);
    });

    it('should not render', () => {
      expect(productCardWrapper.find('.product-card')).toHaveLength(0);
    });
  });
});
