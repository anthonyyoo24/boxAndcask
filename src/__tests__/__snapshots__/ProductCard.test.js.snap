// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard With props matches the snapshot 1`] = `
<div
  className="product-card"
  key="2"
>
  <Link
    to="/products/2"
  >
    <img
      alt="Josephine Sofa"
      className="product-card__img"
      src="josephine-sofa.jpg"
    />
  </Link>
  <div
    className="product-card__content"
  >
    <Link
      className="product-card__name"
      to="/products/2"
    >
      Josephine Sofa
    </Link>
    <p
      className="product-card__price"
    >
      $
      1200.99
    </p>
  </div>
</div>
`;

exports[`ProductCard With props updates via props 1`] = `
<div
  className="product-card"
  key="2"
>
  <Link
    to="/products/2"
  >
    <img
      alt="Josephine Sofa"
      className="product-card__img"
      src="josephine-sofa.jpg"
    />
  </Link>
  <div
    className="product-card__content"
  >
    <Link
      className="product-card__name"
      to="/products/2"
    >
      Josephine Sofa
    </Link>
    <p
      className="product-card__price"
    >
      $
      1000
    </p>
  </div>
</div>
`;

exports[`ProductCard With props updates via props 2`] = `
<div
  className="product-card"
  key="2"
>
  <Link
    to="/products/2"
  >
    <img
      alt="Josephine Sofa"
      className="product-card__img"
      src="josephine-sofa.jpg"
    />
  </Link>
  <div
    className="product-card__content"
  >
    <Link
      className="product-card__name"
      to="/products/2"
    >
      Josephine Sofa
    </Link>
    <p
      className="product-card__price"
    >
      $
      1100.99
    </p>
  </div>
</div>
`;
